azure-storage-blob/azure/storage/blob/_upload_chunking.py

```diff
@@ -211,15 +220,14 @@ def process_chunk(self, chunk_data):
         return self._upload_chunk_with_progress(chunk_offset, chunk_bytes)
 
     def _update_progress(self, length):
-        if self.progress_callback is not None:
-            if self.progress_lock is not None:
-                with self.progress_lock:
-                    self.progress_total += length
-                    total = self.progress_total
-            else:
+        if self.progress_lock is not None:
+            with self.progress_lock:
                 self.progress_total += length
-                total = self.progress_total
-            self.progress_callback(total, self.blob_size)
+        else:
+            self.progress_total += length
+
+        if self.progress_callback is not None:
```

Comment:
hmmm maybe it's better to check near line 242? 

Currently progress is only needed if there's a callback, this change would always keep track of progress, which is additional computation that is not mandatory.